backend/siarnaq/api/episodes/models.py

```diff
@@ -338,3 +431,96 @@ class Meta:
 
     def __str__(self):
         return f"{self.tournament} ({self.name})"
+
+    def enqueue(self):
+        """Creates and enqueues all matches for this round.
+        Fails if this round is already in progress."""
+
+        tournament = challonge.get_tournament(self.tournament.challonge_id_private)
+        # Derive matches of this round
+        # NOTE this probably makes more sense (efficiency and consistency)
+        # as a dict. Track in #549
+        matches = []
+        # Takes some wrangling with API response format
+        # We should move this block later
+        # (to keep all code that directly hits challonge
+        # in its own module) Track in #549
+        for item in tournament["included"]:
+            if item["type"] == "match":
+                round_idx = item["attributes"]["round"]
+                if round_idx == self.challonge_id:
+                    # Only enqueue the round if all matches are "open".
+                    # TODO handle ability to requeue a match
+                    # TODO handle ability to requeue an entire round
+                    if item["attributes"]["state"] != "open":
+                        # TODO return some sort of 400, prob 409
+                        # rather than 500-ing
+                        raise Exception
+                    matches.append(item)
+
+        # Map participant "objects" with IDs for easy lookup
+        participants = dict()
+        for item in tournament["included"]:
+            if item["type"] == "participant":
+                id = item["id"]
+                participants[id] = item
```

Comment:
Here, and similarly for `round_idx` above:

```suggestion
            match item:
                case {"type": "participant", "id": id}:
                    participants[id] = item
```